prereise/gather/demanddata/transportation_electrification/immediate.py

```diff
@@ -0,0 +1,298 @@
+
+    elif veh_type.lower() == "ldt":
+        newdata = mileage.remove_ldv(mileage.load_data(census_region))
+
+    # updates the weekend and weekday values in the nhts data
+    newdata = mileage.update_if_weekend(newdata)
+
+    # add new columns to newdata to store data that is not in NHTS data
+    newdata = newdata.reindex(
+        list(newdata.columns)
+        + [
+            "trip start battery charge",
+            "trip end battery charge",
+            "charging power",
+            "charging time",
+            "charging consumption",
+            "BEV could be used",
+            "trip number",
+        ],
+        axis="columns",
+        fill_value=0,
+    )
+
+    input_day = mileage.get_input_day(mileage.get_model_year_dti(model_year))
+
+    TRANS_charge = np.zeros(24 * len(input_day))
+    data_day = mileage.get_data_day(newdata)
+
+    trip_strategy = 1
+    kwh = kwhmi * veh_range
+    weekday_electricload, weekend_electricload = np.zeros(4800), np.zeros(4800)
+
+    l = len(newdata)
+
+    # weekday first
+    trip_num = 1
+    start_point = 0
+    end_point = 0
+
+    for i in range(l):
+        if data_day[i] == 2:  # 2 is weekday
+            if (
+                i > 0
+                and newdata.iloc[i, newdata.columns.get_loc("sample vehicle number")]
+                == newdata.iloc[i - 1, newdata.columns.get_loc("sample vehicle number")]
+            ):
+                newdata.iloc[
+                    i, newdata.columns.get_loc("trip start battery charge")
+                ] = (
+                    newdata.iloc[
+                        i - 1, newdata.columns.get_loc("trip end battery charge")
+                    ]
+                    + newdata.iloc[
+                        i - 1, newdata.columns.get_loc("charging consumption")
+                    ]
+                )
+                trip_num += 1
+                newdata.iloc[i, newdata.columns.get_loc("trip number")] = trip_num
+
+            else:
+                newdata.iloc[
+                    i, newdata.columns.get_loc("trip start battery charge")
+                ] = kwh
+                trip_num = 1
+                newdata.iloc[i, newdata.columns.get_loc("trip number")] = trip_num
+
+            if (
+                newdata.iloc[i, newdata.columns.get_loc("total vehicle miles traveled")]
+                < veh_range * const.ER
+            ):
+                newdata.iloc[i, newdata.columns.get_loc("BEV could be used")] = 1
+                newdata.iloc[i, newdata.columns.get_loc("trip end battery charge")] = (
+                    newdata.iloc[
+                        i, newdata.columns.get_loc("trip start battery charge")
+                    ]
+                    - newdata.iloc[i, newdata.columns.get_loc("Miles traveled")]
+                    * kwhmi
+                    * const.ER
+                )
+            else:
+                newdata.iloc[i, newdata.columns.get_loc("BEV could be used")] = 0
```

Comment:
Once we have the values of `"trip end battery charge"`, can we calculate the values for the new `"BEV could be used"` and `"trip end battery charge"` columns for all columns at once, rather than within a loop? E.g. something like:
```python
newdata["BEV could be used"] = (newdata["total vehicle miles traveled"] < veh_range * const.ER).map(int)
newdata.loc[newdata["BEV could be used"] == 1, "trip end battery charge"] = (
    newdata.loc[newdata["BEV could be used"] == 1, "trip start battery charge"]
    - newdata.loc[newdata["BEV could be used"] == 1, "Miles traveled"]
    * kwhmi
    * const.ER
)
```
If we keep the `"BEV could be used"` column as a boolean, it gets even simpler, since we can use the boolean column directly for slicing:
```python
newdata["BEV could be used"] = newdata["total vehicle miles traveled"] < veh_range * const.ER
newdata.loc[newdata["BEV could be used"], "trip end battery charge"] = (
    newdata.loc[newdata["BEV could be used"], "trip start battery charge"]
    - newdata.loc[newdata["BEV could be used"], "Miles traveled"]
    * kwhmi
    * const.ER
)
```